contracts/strategies/StrategyDAI3pool.sol

```diff
@@ -0,0 +1,334 @@
+            flag = false;
+        }
+
+        rebalance();
+        uint256 _want = (IERC20(want).balanceOf(address(this))).sub(tank);
+        if (_want > 0) {
+            IERC20(want).safeApprove(_3pool, 0);
+            IERC20(want).safeApprove(_3pool, _want);
+            uint256 v = _want.mul(1e18).div(ICurveFi(_3pool).get_virtual_price());
+            ICurveFi(_3pool).add_liquidity([_want, 0, 0], v.mul(DENOMINATOR.sub(slip)).div(DENOMINATOR));
+        }
+        uint256 _bal = IERC20(_3crv).balanceOf(address(this));
+        if (_bal > 0) {
+            IERC20(_3crv).safeApprove(y3crv, 0);
+            IERC20(_3crv).safeApprove(y3crv, _bal);
+            yvERC20(y3crv).deposit(_bal);
+        }
+    }
+
+    // Controller only function for creating additional rewards from dust
+    function withdraw(IERC20 _asset) external returns (uint256 balance) {
+        require(msg.sender == controller, "!controller");
+        require(want != address(_asset), "want");
+        require(_3crv != address(_asset), "3crv");
+        require(y3crv != address(_asset), "y3crv");
+        balance = _asset.balanceOf(address(this));
+        _asset.safeTransfer(controller, balance);
+    }
+
+    // Withdraw partial funds, normally used with a vault withdrawal
+    function withdraw(uint256 _amount) external {
+        require(msg.sender == controller, "!controller");
+
+        rebalance();
+        uint256 _balance = IERC20(want).balanceOf(address(this));
+        if (_balance < _amount) {
+            _amount = _withdrawSome(_amount.sub(_balance));
+            _amount = _amount.add(_balance);
+            tank = 0;
+        } else {
+            if (tank >= _amount) tank = tank.sub(_amount);
+            else tank = 0;
+        }
+
+        address _vault = IController(controller).vaults(address(want));
+        require(_vault != address(0), "!vault"); // additional protection so we don't burn the funds
+        uint256 _fee = _amount.mul(withdrawalFee).div(DENOMINATOR);
+        IERC20(want).safeTransfer(IController(controller).rewards(), _fee);
+        IERC20(want).safeTransfer(_vault, _amount.sub(_fee));
+    }
+
+    function _withdrawSome(uint256 _amount) internal returns (uint256) {
+        uint256 _amnt = _amount.mul(1e18).div(ICurveFi(_3pool).get_virtual_price());
+        uint256 _amt = _amnt.mul(1e18).div(yvERC20(y3crv).getPricePerFullShare());
+        uint256 _before = IERC20(_3crv).balanceOf(address(this));
+        yvERC20(y3crv).withdraw(_amt);
+        uint256 _after = IERC20(_3crv).balanceOf(address(this));
+        return _withdrawOne(_after.sub(_before));
+    }
+
+    function _withdrawOne(uint256 _amnt) internal returns (uint256) {
+        uint256 _before = IERC20(want).balanceOf(address(this));
+        IERC20(_3crv).safeApprove(_3pool, 0);
+        IERC20(_3crv).safeApprove(_3pool, _amnt);
+        ICurveFi(_3pool).remove_liquidity_one_coin(_amnt, 0, _amnt.mul(DENOMINATOR.sub(slip)).div(DENOMINATOR));
```

Comment:
Here in 3PoolUSDT strat you used another division in addition to the `DENOMINATOR`

Just wanted to confirm that here not using another extra div is expected
https://github.com/iearn-finance/yearn-protocol/blob/568906a17e665a20368755010c4cf0a4eaa3faef/contracts/strategies/StrategyUSDT3pool.sol#L137

Reply:
yes. it's because USDT is 6 decimals but DAI is 18.